### ISSUE TITLE
With maint-1.0, build error in MPAS on pm-cpu/intel

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/E3SM#5709]
After PR https://github.com/E3SM-Project/E3SM/pull/5696 to maint-1.0, we can work with pm-cpu, however, there are still some issues requiring changes in submodules.

To use Intel, we need a work-around for this build error:

```
xml_stream_parser.c:1732:6: error: call to undeclared function 'stream_mgr_add_field_c'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                                        stream_mgr_add_field_c(manager, streamID, (const char *)fieldname, packages_local, &err);
```

I am able to simply copy the current `xml_stream_parser.c` from master and it works, but there may be a better fix

**Comment by jonbob** [comment]
@ndkeen -- I think we were working on this in [MPAS-Dev/MPAS-Model PR #1048](https://github.com/MPAS-Dev/MPAS-Model/pull/1048). It looks like this never got in?

**Comment by ndkeen** [comment]
I was trying there, but I understand it might not ideal fix. I just wanted to make a separate issue after the PR to allow working with pm-cpu.

**Comment by rljacob** [comment]
@jonbob can you please make a PR on maint-1.0 to add the correct fix?

**Comment by jonbob** [comment]
@rljacob - yes, I'll work on that and hopefully get it done today. I'm off tomorrow, so otherwise it will be Wednesday

**Comment by rljacob** [comment]
Fixed by #5722
